src/stores/UserStore.ts

```diff
@@ -89,6 +94,28 @@ export const useUserStore = defineStore('UserStore', () => {
     return _userStoreSession.value.env ?? '';
   });
 
+  const jiraUsername = computed(() => {
+    return _userStoreSession.value.jiraUsername ?? '';
+  });
+
+  async function cacheJiraUsername() {
+    if (_userStoreSession.value.jiraUsername === '') {
+      try {
+        const res = await getJiraUserInfo();
+        if (res.response.ok) {
+          _userStoreSession.value.jiraUsername = res.data.name;
+        }
+      } catch (error) {
+        addToast({
+          'title': 'Jira Error',
+          'body': `Error getting your username from <a href="${osimRuntime.value.backends.jiraDisplay}">Jira</a>. ` +
+          'Is your Jira Access Token valid?',
+          'bodyHtml': true,
+          'css': 'warning',
+        });
+      }
+    }
+  }
```

Comment:
Is not the `cacheJiraUsername` expendable? For example having the logic of checking/getting the Jira username on `jiraUsername`?

Regarding the error handling I think is preferable to do it in the service, cause it separates the concerns between the store stuff and the fetching logic.
If that's the case may be it also have sense to make the error message more generic, so if other component uses a Jira username fetch from the service the message is also valid even if the context in the application is different.

Reply:
Is it possible to have a async computed, I read somewhere that async computed is only working with some plugin. I was actually initially thinking about caching the jiraUsername when accessing it in the storage, but I wasn't sure whether it should do such side effects, wdyt?

Agreed about the error message, will move it to the Jira service.